packages/mui-joy/src/Autocomplete/Autocomplete.tsx

```diff
@@ -51,8 +51,8 @@ type OwnerState = Omit<AutocompleteOwnerState<any, any, any, any>, 'onChange' |
 
 const defaultIsActiveElementInListbox = (listboxRef: React.RefObject<HTMLElement>) =>
   listboxRef.current !== null && listboxRef.current.contains(document.activeElement);
-const defaultGetOptionLabel = <T extends unknown>(option: T) =>
-  (option as { label: string }).label ?? option;
+// @ts-ignore
+const defaultGetOptionLabel = (option) => option.label ?? option;
```

Comment:
Is this intended?

Reply:
Yes, otherwise, we get an error that requires us to change type `getOptionLabel` in `AutocompleteProps` to match this